src/client/ds3/pool.c

```diff
@@ -94,7 +94,7 @@ ds3_connect(const char *pool, const char *sys, ds3_t **ds3, daos_event_t *ev)
 		return -ENOMEM;
 
 	/** Copy pool name */
-	strcpy(ds3_tmp->pool, pool);
+	strncpy(ds3_tmp->pool, pool, DAOS_PROP_LABEL_MAX_LEN);
```

Comment:
Could be useful to check the size returned by `strncpy()`. In case of error, an `\0` should be put at the end of the copied string and a log a warning message indicating that the pool label has been truncated.

Reply:
sure.

Reply:
note that strncpy does not return a size. it just copies up to that length and does not null terminate if the length overflows. I updated to just add a null terminator at the end to handle the overflow case (which is what we do in other APIs).

Reply:
Oups, mixed with strnlen 🙇 
Fine for me .